src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -32,6 +34,12 @@
      */
     public abstract void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result);
 
+    /**
+     * Looks at the definition of {@link AbstractProject} and list up its branches, then puts them into
+     * the collection.
+     */
+    public abstract void parseAssociatedBranches(AbstractProject<?,?> job, Collection<GitHubBranch> result);
```

Comment:
` public abstract class GitHubRepositoryNameContributor implements ExtensionPoint `
Breaking compatibility for ExtensionPoint is bad idea.